mlflow/pyfunc/scoring_server/utils.py

```diff
@@ -0,0 +1,38 @@
+import logging
+from typing import Any, Dict, Union
+
+import mlflow
+from mlflow.exceptions import MlflowException
+from mlflow.pyfunc.scoring_server import _parse_json_data
+
+_logger = logging.getLogger(__name__)
+
+
+# NB: this function should always be kept in sync with the serving
+# process in scoring_server invocations.
+def _validate_serving_input(model_uri: str, serving_input: Union[str, Dict[str, Any]]):
+    """
+    Helper function to validate the model can be served and provided input is valid
+    prior to serving the model.
+
+    Args:
+        model_uri: URI of the model to be served.
+        serving_input: Input data to be validated. Should be a dictionary or a JSON string.
+
+    Returns:
+        The prediction result from the model.
+    """
+
+    try:
+        # sklearn model might not have python_function flavor if it
+        # doesn't define a predict function. In such case the model
+        # can not be served anyways
+        pyfunc_model = mlflow.pyfunc.load_model(model_uri)
```

Comment:
Can we clean up the model artifacts after validation to save disk space?

Reply:
+1 - this could prove problematic for transformers models

Reply:
Done!